py-polars/tests/unit/io/test_hive.py

```diff
@@ -65,6 +65,26 @@ def test_hive_partitioned_predicate_pushdown(
     )
 
 
+@pytest.mark.write_disk()
+def test_hive_partitioned_predicate_pushdown_skips_correct_number_of_files(
+    io_files_path: Path, tmp_path: Path, monkeypatch: Any, capfd: Any
+) -> None:
+    monkeypatch.setenv("POLARS_VERBOSE", "1")
+    df = pl.DataFrame({"d": pl.arange(0, 10_000, eager=True)}).with_columns(
```

Comment:
Can you ensure this test doesn't write so much data? Try the minimal amount of rows need to replicate the behavior.